### ISSUE TITLE
AttributeError: __exit__ while deleting directory

### ISSUE THREAD

**Issue by ediowar** [bleachbit/bleachbit#810]
Ubuntu 16.04.6 LTS, BleachBit version 3.9.0
Running from source code, Git revision https://github.com/bleachbit/bleachbit/commit/047c4c5ceb10999a1132bfa8128b0390e8fe8f41
The box for "Overwrite contents of files to prevent recovery" in the preferences is checked.
```
Error: system.custom: Command to delete /home/dzmitry/**EVERY DIRECTORY TO DELETE**
Traceback (most recent call last):
  File "/home/dzmitry/Downloads/bleachbit-master/bleachbit/Worker.py", line 87, in execute
    for ret in cmd.execute(self.really_delete):
  File "/home/dzmitry/Downloads/bleachbit-master/bleachbit/Command.py", line 82, in execute
    FileUtilities.delete(self.path, self.shred)
  File "/home/dzmitry/Downloads/bleachbit-master/bleachbit/FileUtilities.py", line 317, in delete
    if not is_dir_empty(path):
  File "/home/dzmitry/Downloads/bleachbit-master/bleachbit/FileUtilities.py", line 614, in is_dir_empty
    with os.scandir(dirname) as it:
AttributeError: __exit__
```
The number of errors is equal to the number of folders to delete.

**Comment by az0** [comment]
You wrote you have Git revision 047c4c5, but this looks like an error that was fixed in Git revision e2642d19178b3811109cc015a45d7262e3e4a469. One reason I think you have an older revision is the traceback shows line 614 as `with os.scandir(dirname) as it`, but now the line is `for entry in os.scandir(dirname):`.

Would you please double check you have the latest version?

**Comment by ediowar** [comment]
You're right, this is an older revision.
